framework/src/modules/interoperability/sidechain/module.ts

```diff
@@ -195,4 +211,184 @@ export class SidechainInteroperabilityModule extends BaseInteroperabilityModule
+				);
+			}
+			// CAUTION!
+			// this check is intentionally applied after MIN_CHAIN_NAME_LENGTH, as it will fail for empty string
+			if (!isValidName(ownChainName)) {
+				throw new Error(`ownChainName must have only ${validNameCharset} character set.`);
+			}
+			if (ownChainName === CHAIN_NAME_MAINCHAIN) {
+				throw new Error(`ownChainName must be not equal to ${CHAIN_NAME_MAINCHAIN}.`);
+			}
+
+			// ownChainNonce > 0
+			if (ownChainNonce < 1) {
+				throw new Error('ownChainNonce must be > 0.');
+			}
+
+			// chainInfos contains exactly one entry mainchainInfo with:
+			if (chainInfos.length !== 1) {
+				throw new Error('chainInfos must contain exactly one entry.');
+			}
+			// mainchainInfo.chainID == getMainchainID();
+			const mainchainInfo = chainInfos[0];
+			const mainchainID = getMainchainID(mainchainInfo.chainID);
+			if (!mainchainInfo.chainID.equals(mainchainID)) {
+				throw new Error(`mainchainInfo.chainID must be equal to ${mainchainID.toString('hex')}.`);
+			}
+			// mainchainInfo.chainData.name == CHAIN_NAME_MAINCHAIN,
+			// mainchainInfo.chainData.status is either equal to CHAIN_STATUS_REGISTERED or to CHAIN_STATUS_ACTIVE,
+			// mainchainInfo.chainData.lastCertificate.timestamp < g.header.timestamp;
+			if (mainchainInfo.chainData.name !== CHAIN_NAME_MAINCHAIN) {
+				throw new Error(`chainData.name must be equal to ${CHAIN_NAME_MAINCHAIN}.`);
+			}
+			const validStatuses = [ChainStatus.REGISTERED, ChainStatus.ACTIVE];
+			if (!validStatuses.includes(mainchainInfo.chainData.status)) {
+				throw new Error(`chainData.status must be one of ${validStatuses.join(', ')}.`);
+			}
+			if (mainchainInfo.chainData.lastCertificate.timestamp > ctx.header.timestamp) {
+				throw new Error('chainData.lastCertificate.timestamp must be < header.timestamp.');
+			}
+			// channelData
+			this._verifyChannelData(ctx, mainchainInfo);
+
+			// activeValidators
+			this._verifyChainValidators(mainchainInfo);
+		}
+
+		// terminatedStateAccounts
+		await this._verifyTerminatedStateAccounts(ctx, terminatedStateAccounts);
+
+		// terminatedOutboxAccounts
+		if (terminatedOutboxAccounts.length !== 0) {
+			throw new Error('terminatedOutboxAccounts must be empty.');
+		}
+	}
+
+	private async _verifyTerminatedStateAccounts(
+		ctx: GenesisBlockExecuteContext,
+		terminatedStateAccounts: TerminatedStateAccountWithChainID[],
+	) {
+		this._verifyTerminatedStateAccountsCommon(terminatedStateAccounts);
+
+		const mainchainID = getMainchainID(ctx.chainID);
+		const ownChainAccount = await this.stores.get(OwnChainAccountStore).get(ctx, EMPTY_BYTES);
+
+		for (const stateAccount of terminatedStateAccounts) {
+			// stateAccount.chainID != getMainchainID()
+			if (stateAccount.chainID.equals(mainchainID)) {
+				throw new Error(
+					`stateAccount.chainID must not be equal to ${mainchainID.toString('hex')}.`,
+				);
+			}
+
+			// and stateAccount.chainID != ownChainAccount.chainID.
```

Comment:
I guess this check should be moved in the `verifyTerminatedStateAccountsCommon`, and should be performed for both mainchain and sidechain cases. I will update the LIP accordingly.